File: Java/RecursionPractice/src/calculations/Fibonacci.java
package calculations;

public class Fibonacci {

    int index;

    public int getValue() {
        return this.recursivelyGetValue(this.index);
    }

    private int recursivelyGetValue(int mutableIndex) {
        if (mutableIndex < 2) {
            return mutableIndex;
        }
        return this.recursivelyGetValue(mutableIndex - 1)
                + this.recursivelyGetValue(mutableIndex - 2);
    }

    public Fibonacci(int index) {
        this.index = index;
    }
}

File: Kotlin/RecursionPractice/src/calculations/Fibonacci.kt
package calculations

class Fibonacci(private val index : Int) {

    private fun recursivelyCalculate(descendingIndex : Int): Int {
        if (descendingIndex == 0 || descendingIndex == 1) {
            return descendingIndex
        }
        return recursivelyCalculate(descendingIndex - 1) + recursivelyCalculate(descendingIndex -2)
    }

    fun getValue(): Int {
        return recursivelyCalculate(index);
    }
}